chapter06/spg/exercise2.pl

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use JSON;
+
+my $output = "stored_data.json";
+
+my $data = {};
+
+if ( -s $output ) {
+  open( my $input_file, '<', $output ) or die "could not create file '$output' $!";
+  ;
+  my $json_string;
+  while ( <$input_file> ) {
+    $json_string .= $_;
+  }
```

Comment:
For a file that isn't too big, you could use the glob in array context to get all the lines at once and then join them.